packages/shared/src/components/PostOptionsReadingHistoryMenu.tsx

```diff
@@ -0,0 +1,141 @@
+import { MenuIcon } from './MenuIcon';
+import { QueryIndexes } from '../hooks/useReadingHistory';
+
+const PortalMenu = dynamic(() => import('./fields/PortalMenu'), {
+  ssr: false,
+});
+
+export type PostOptionsReadingHistoryMenuProps = {
+  post: ReadHistoryPost;
+  onHidden?: () => unknown;
+  displayCopiedMessageFunc?: () => void;
+  onHide?: (postId: string) => unknown;
+  indexes: QueryIndexes;
+};
+
+/* eslint-disable no-param-reassign */
+const updateReadingHistoryPost =
+  (
+    queryClient: QueryClient,
+    historyQueryKey: QueryKey,
+    readHistoryPostUpdated: Partial<ReadHistoryPost>,
+    { page, edge }: QueryIndexes,
+  ): ((args: { id: string }) => Promise<() => void>) =>
+  async () => {
+    const oldReadingHistory =
+      queryClient.getQueryData<ReadHistoryInfiniteData>(historyQueryKey);
+    const newItems = [...oldReadingHistory.pages];
+    const history = newItems[page].readHistory.edges[edge].node.post;
+    newItems[page].readHistory.edges[edge].node.post = {
+      ...history,
+      ...readHistoryPostUpdated,
+    };
+    queryClient.setQueryData<ReadHistoryInfiniteData>(historyQueryKey, {
+      ...oldReadingHistory,
+      pages: newItems,
+    });
+    return () => {
+      queryClient.setQueryData<ReadHistoryInfiniteData>(
+        historyQueryKey,
+        oldReadingHistory,
+      );
+    };
+  };
+
+export default function PostOptionsReadingHistoryMenu({
+  post,
+  onHidden,
+  onHide,
+  indexes,
+}: PostOptionsReadingHistoryMenuProps): ReactElement {
+  const { user } = useContext(AuthContext);
+  const queryClient = useQueryClient();
+  const historyQueryKey = ['readHistory', user?.id];
+
+  const [, copyPostLink] = useCopyLink(() => post.commentsPermalink);
+
+  const { bookmark, removeBookmark } = useBookmarkPost({
+    onBookmarkMutate: updateReadingHistoryPost(
+      queryClient,
+      historyQueryKey,
+      {
+        bookmarked: true,
+      },
+      indexes,
+    ),
+    onRemoveBookmarkMutate: updateReadingHistoryPost(
+      queryClient,
+      historyQueryKey,
+      {
+        bookmarked: false,
+      },
+      indexes,
+    ),
+  });
+
+  const onBookmarkReadingHistoryPost = async (): Promise<void> => {
+    if (post?.bookmarked) {
+      return removeBookmark(post);
+    }
+    return bookmark(post);
+  };
+
+  const getBookmarkIcon = () => {
+    if (post?.bookmarked) {
+      return <MenuIcon Icon={BookmarkIconFilled} />;
+    }
+    return <MenuIcon Icon={BookmarkIcon} />;
+  };
+
+  const getBookmarkMenuText = () => {
+    if (post?.bookmarked) {
+      return 'Remove from bookmarks';
+    }
+    return 'Save to bookmarks';
+  };
+
+  return (
+    <>
+      <PortalMenu
+        disableBoundariesCheck
+        id="reading-history-options-context"
+        className="menu-primary"
+        animation="fade"
+        onHidden={onHidden}
+      >
+        <Item className="typo-callout" onClick={onBookmarkReadingHistoryPost}>
+          <a className="flex w-full typo-callout">
```

Comment:
Why do you use an anchor element here?

Reply:
@idoshamun I reused it from our PostOptionsMenu. Should we use something else?

Reply:
Shouldn't be there as well I think. Anchor elements are for links not for general-purpose buttons. I think the `Item` component is enough

Reply:
You are right, the anchor is not needed. I still have to use span to enforce the styles.

Reply:
Why can't you apply them on the `Item` directly?

Reply:
It seems that the text style in the Item is ignored if I do it.

Reply:
It's worth checking the `Item` component itself in this case

Reply:
@idoshamun We have to wrap something inside the `Item` if another typo is needed.
We set the inner div with `typo-footnote` by default (CSS)

So the only way to use other fonts is to override it.
We might have to look into this if it's still needed actually. or it should be `typo-callout` by default.
But suggestion we open a separate ticket for that.

Reply:
So let's use a `span` then as suggested by @ayadotdev